src/http/url.cc

```diff
@@ -68,6 +81,35 @@ bool url_decode(const std::string_view& in, sstring& out) {
     return true;
 }
 
+sstring url_encode(const std::string_view& in) {
+    size_t encodable_chars = 0;
+    for (size_t i = 0; i < in.length(); i++) {
+        if (should_encode(in[i])) {
+            encodable_chars++;
+        }
+    }
+
+    if (encodable_chars == 0) {
+        return sstring(in);
+    }
+
+    sstring ret(in.length() + encodable_chars * 2 + 1, 0);
```

Comment:
Why is the "+1" needed here in the length?

Reply:
For trailing zero. Maybe I'm wrong, but this constructor defines the raw buffer size, but in.length() doesn't include it